setup.py

```diff
@@ -8,6 +8,15 @@
 
 CURRENT_DIR = os.path.dirname(__file__)
 
+def get_lib_path():
+    """Get library path, name and version"""
+    # Directly exec libinfo to get the right setup
+    libinfo_py = os.path.join(CURRENT_DIR, "./python/mlc_chat/libinfo.py")
```

Comment:
From here to the `exec(compile(...))` line looks very similar to the start of `git_describe_version`.  Can we pull it out into a separate function that accepts a path and returns the evaluated module?

(If you'd like, we could pull in a version I wrote for a hobby project ([link](https://github.com/Lunderberg/installation_scripts/blob/master/pybuild#L30-L47)), which does so using the `importlib` machinery.)